src/DIRAC/WorkloadManagementSystem/JobWrapper/JobWrapper.py

```diff
@@ -30,6 +30,8 @@
 import distutils.spawn
 import six
 
+if six.PY2:
+    from codecs import open
```

Comment:
Is there a reason for using `codecs` rather than `io` here?
```suggestion
    from io import open
```

Reply:
We tried that (I think, it's all a blur) and it didn't work and I hope that @sfayer remembers what the problem was.

Reply:
Ah yes, I'd forgotten about this:

```python
>>> with io.open("/tmp/test.txt", "a+", encoding="utf-8") as outputFile:
...     outputFile.write("line")
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
TypeError: write() argument 1 must be unicode, not str
```

It's Python 2 only so 🤷

Reply:
The problem is that the io version of open can't cope with writing str types on python2 if encoding is set (which isn't the same behaviour as python3):
```
Python 2.7.13 (default, Feb 24 2021, 11:28:05)
>>> fd = open("test.out", "w", encoding="utf-8")
>>> fd.write("test")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: write() argument 1 must be unicode, not str
```
```
Python 3.9.10 | packaged by conda-forge | (main, Feb  1 2022, 21:24:11) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> fd = open("test.out", "w", encoding="utf-8")
>>> fd.write("test")
4
```

The codecs version of open doesn't seem to have this issue...

Reply:
Does this actually work for unicode data?
```python
>>> print(xxx)
🎉
>>> with codecs.open("/tmp/test.txt", "a+", encoding="utf-8") as outputFile:
...     outputFile.write(xxx)
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/usr/lib/python2.7/codecs.py", line 708, in write
    return self.writer.write(data)
  File "/usr/lib/python2.7/codecs.py", line 369, in write
    data, consumed = self.encode(object, self.errors)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xf0 in position 0: ordinal not in range(128)
```

Reply:
Yes, it works if type(xxx) is unicode but not if it's a str containing unicode (as it assumes ascii for the encode in that case).

In this specific instance the "line" variable that's written does always seem to be of unicode type when there are unicode characters present (at least on python2): We tested this with the full pilot running test jobs with every situation of unicode and non-unicode output we could think of and it worked as expected in every case.